src/WorkBC.Data/Migrations/20240413000723_InsertNocCodes2021.cs

```diff
@@ -0,0 +1,136 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Migrations;
+using System.Text.Json;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System;
+using WorkBC.Data.Model.JobBoard;
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
+using WorkBC.Data.Model.Enterprise;
+using System.IO;
+using Microsoft.VisualBasic;
+using System.Text;
+using System.Linq;
+
+#nullable disable
+
+namespace WorkBC.Data.Migrations
+{
+    public partial class InsertNocCodes2021 : Migration
+    {
+        private readonly HttpClient _httpClient;
+        private readonly string _ssotUrl;
+
+        public InsertNocCodes2021() : base()
+        {
+            _httpClient = new HttpClient();
+            _ssotUrl = Environment.GetEnvironmentVariable("SSOT_URL");
+            if (_ssotUrl == null)
+            {
+                throw new Exception("SSOT_URL not set");
+            }
+        }
+        protected async override void Up(MigrationBuilder migrationBuilder)
+        {
+            Console.WriteLine("Hello");
+            // Append to NocCodes2021  table from SSOT API
+            var allNocs = await GetNocCodes2021(_ssotUrl);
+
+            Console.WriteLine("allNocs: " + allNocs.Count.ToString());
+            Console.WriteLine("Inside Up method");
+
+            //migrationBuilder.Sql(@"INSERT INTO [dbo].[NocCodes2021]
+            //           ([Id],[Code],[Title], [FrenchTitle])
+            //     VALUES
+            //           (1, '1', 'test1, test2', 'testFr1')");
+
+
+            foreach (var nocs in allNocs)
+            {
+
+                if (String.IsNullOrEmpty(nocs.noc_2021))
+                {
+                    Console.WriteLine("null noc");
```

Comment:
I think this should be an Exception instead.

Reply:
Removed the try-catch blocks in the latest commit.